cmd/controller/run.go

```diff
@@ -194,14 +197,15 @@ func Run(opts Options, runLog logr.Logger) error {
 	}
 	{ // add controller for apps
 		appFactory := app.CRDAppFactory{
-			CoreClient:  coreClient,
-			AppClient:   kcClient,
-			KcConfig:    kcConfig,
-			AppMetrics:  appMetrics,
-			CmdRunner:   sidecarCmdExec,
-			Kubeconf:    kubeconf,
-			CompInfo:    compInfo,
-			CacheFolder: cacheFolderApps,
+			CoreClient:           coreClient,
+			AppClient:            kcClient,
+			KcConfig:             kcConfig,
+			CountMetrics:         countMetrics,
+			ReconcileTimeMetrics: reconcileTimeMetrics,
```

Comment:
Do we expect to have different types of metrics in the future? If yes, then we should probably move these into a common struct so that it's easier to pass them around.

Reply:
Same for pkgi

Reply:
so you mean should a single struct for both the metrics in pkg/metrics and put both of them in a single file instead of two different files as of now?